internal/pkg/instrumentation/bpf/google.golang.org/grpc/client/bpf/probe.bpf.c

```diff
@@ -105,7 +108,34 @@ int uprobe_ClientConn_Invoke(struct pt_regs *ctx)
     return 0;
 }
 
-UPROBE_RETURN(ClientConn_Invoke, struct grpc_request_t, grpc_events, events, 3, 0, true)
+// This instrumentation attaches uprobe to the following function:
+// func (cc *ClientConn) Invoke(ctx context.Context, method string, args, reply interface{}, opts ...CallOption) error
+SEC("uprobe/ClientConn_Invoke")
+int uprobe_ClientConn_Invoke_Returns(struct pt_regs *ctx) {
+    struct go_iface go_context = {0};
+    get_Go_context(ctx, 3, 0, true, &go_context);
+    void *key = get_consistent_key(ctx, go_context.data);
+    struct grpc_request_t *grpc_span = bpf_map_lookup_elem(&grpc_events, &key);
+    if (grpc_span == NULL) {
+        bpf_printk("event is NULL in ret probe");
+        return 0;
+    }
+
+    // Getting the returned response (error)
+    void *resp_ptr = get_argument(ctx, 1);
+    void *status_ptr = 0;
+    // get `s` field from Status object pointer (from returned error interface)
+    // (*Status)error{*s}
+    bpf_probe_read(&status_ptr, sizeof(status_ptr), (void *)(resp_ptr+status_s_pos));
+    // Get status code from Status.s pointer
+    // (*s)struct{Code}
+    bpf_probe_read(&grpc_span->status_code, sizeof(grpc_span->status_code), (void *)(status_ptr + status_code_pos));
```

Comment:
I think the problem is in this line:
`void *resp_ptr = get_argument(ctx, 1);`
Since we are interested in the data pointer of the interface, we should read the second returned argument (the firs one is the `itab` pointer)

https://github.com/golang/go/blob/aeac0b6cbfb42bc9c9301913a191bb09454d316a/src/runtime/runtime2.go#L178

Reply:
Thanks, so would that be `void *resp_ptr = get_argument(ctx, 2)`? I couldn't figure out how we determine the argument index in the other ebpf functions, so I tried brute-forcing this with 1-9 and got the same kind of results

Reply:
Yes, other than that the code looks fine to me.
* When reading the `status_ptr` worth checking that the pointer we read is not NILL (if it is, avoid reading the code). I assume that in a happy case, a nil error will be returned.
* Maybe worth considering using `bpf_probe_read_user` instead of `bpf_probe_read` as it is more recommended for these cases but I'm not sure it will make a difference in terms of correctness.

Reply:
Hm still no luck, I tried checking the 2nd argument and using bpf_probe_read_user in https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1044/commits/540ff896453c922afe3e30066f64fdeacdfc0d88 but that's just getting 0 for both success and error. I logged the error message from the test app just to be sure and it is returning "Unavailable" which should be gRPC code `14`

Reply:
Try and print the status code value in epbf using `bpf_printk`. maybe the int value is not getting read properly by the go code (alignment might be an issue)